go/vt/vttablet/tabletserver/tabletserver.go

```diff
@@ -1867,6 +1867,13 @@ func (tsv *TabletServer) BroadcastHealth(terTimestamp int64, stats *querypb.Real
 // HeartbeatLag returns the current lag as calculated by the heartbeat
 // package, if heartbeat is enabled. Otherwise returns 0.
 func (tsv *TabletServer) HeartbeatLag() (time.Duration, error) {
+	// If the reader is closed and we are not serving, then the
+	// query service is shutdown and this value is not being updated.
+	// We return healthy from this as a signal to the healtcheck to attempt
+	// to start the query service again
+	if !tsv.hr.IsOpen() && !tsv.IsServing() {
+		return 0, nil
```

Comment:
Is it possible to return something to indicate the reader is closed?

i.e. -1 or max int

Reply:
We could return -1, if we return max int, then we will get an unhealthy due to replication lag issue. We could also return a custom error that specifies the reader is closed. We would need to add code to handle -1 or the custom error inside the healthcheck.

Reply:
Given that this is an improvement already, I'll merge this. We can fix forward as needed.